src/Expr.cc

```diff
@@ -1634,14 +1584,18 @@ AddExpr::AddExpr(ExprPtr arg_op1, ExprPtr arg_op2)
 		return;
 
 	TypeTag bt1 = op1->GetType()->Tag();
-
-	if ( IsVector(bt1) )
-		bt1 = op1->GetType()->AsVectorType()->Yield()->Tag();
-
 	TypeTag bt2 = op2->GetType()->Tag();
 
-	if ( IsVector(bt2) )
+	if ( IsVector(bt1) && IsVector(bt2) )
+		{
+		bt1 = op1->GetType()->AsVectorType()->Yield()->Tag();
 		bt2 = op2->GetType()->AsVectorType()->Yield()->Tag();
+		}
+	else if ( IsVector(bt1) || IsVector(bt2) )
+		{
+		ExprError("cannot mix vector and scalar operands");
```

Comment:
There's an opportunity to factor this out in a number of places, might be worth considering.

Reply:
Yes. Added a static helper rather than adding a member that would be part of the interface. It's not pretty, but it's less lines.